tests/diff/float_values_test.go

```diff
@@ -76,13 +159,63 @@ func TestFloatValues(t *testing.T) {
 		}
 	})
 
-	// TODO https://github.com/FerretDB/dance/issues/266
-	/*t.Run("Update", func(t *testing.T) {
+	t.Run("FindAndModify", func(t *testing.T) {
+		t.Parallel()
 
-	})*/
+		for name, tc := range map[string]struct {
+			filter   bson.D
+			update   bson.D
+			expected mongo.CommandError
+		}{
+			"NaN": {
+				filter: bson.D{{"_id", "1"}},
+				update: bson.D{{"$set", bson.D{{"foo", math.NaN()}}}},
+				expected: mongo.CommandError{
+					Code: 2,
+					Name: "BadValue",
+					Message: `wire.OpMsg.Document: validation failed for { findAndModify: ` +
+						`"findAndModify-NaN", query: { _id: "1" }, update: { $set: { foo: nan.0 } }, ` +
+						`$db: "testfloatvalues" } with: NaN is not supported`,
+				},
+			},
+			"NegativeZero": {
+				filter: bson.D{{"_id", "1"}},
+				update: bson.D{{"$set", bson.D{{"foo", math.Copysign(0.0, -1)}}}},
+				expected: mongo.CommandError{
+					Code: 2,
+					Name: "BadValue",
+					Message: `wire.OpMsg.Document: validation failed for { findAndModify: ` +
+						`"findAndModify-NegativeZero", query: { _id: "1" }, update: { $set: { foo: ` +
+						`-0.0 } }, $db: "testfloatvalues" } with: -0 is not supported`,
+				},
+			},
+		} {
+			name, tc := name, tc
+
+			t.Run(name, func(t *testing.T) {
+				t.Parallel()
+
+				// to return error
+				var update any
+
+				err := db.Collection("findAndModify-"+name).FindOneAndUpdate(ctx, tc.filter, tc.update).Decode(update)
+
+				t.Run("FerretDB", func(t *testing.T) {
+					t.Parallel()
 
-	// TODO https://github.com/FerretDB/dance/issues/266
-	/*t.Run("FindAndModify", func(t *testing.T) {
+					assertEqualError(t, tc.expected, err)
+				})
+
+				t.Run("MongoDB", func(t *testing.T) {
+					t.Parallel()
 
-	})*/
+					if err != nil {
+						if !errors.Is(err, mongo.ErrNoDocuments) {
+							t.Fail()
```

Comment:
I suggest to insert some data before updating it, and this this check won't be needed (there is always something to update).